src/backend/routers/dashboard.py

```
@@ -291,19 +292,28 @@ def get_backups() -> Dict[str, Any]:
+    try:
+        manager = get_backup_manager()
+        result = manager.verify_backup_integrity(backup_id)
+        return {
+            "backup_id": backup_id,
+            "valid": result.get("is_valid", False),
+            "integrity_verified": result,
+            "timestamp": datetime.now(timezone.utc).isoformat(),
+        }
+    except ValueError as e:
+        logger.error(f"Failed to verify backup: {e}")
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)